js/chat/chat.ts

```diff
@@ -95,8 +95,38 @@ class ChatMessages extends LightElement {
 }
 
 class ChatInput extends LightElement {
+  private _disabled = false;
+
   @property() placeholder = "Enter a message...";
-  @property({ type: Boolean, reflect: true }) disabled = false;
+  @property({ type: Boolean })
```

Comment:
`reflect: true` doesn't work here because `LightElement` overwrites lit's `createRenderRoot()` that would typically handle reflecting the property. I think it's worth doing ourselves, which also gives us a place in `ChatInput` to handle the disabled lifecycle so we don't miss anything (e.g. the button state reflecting the global disabled state).

Reply:
Might be worth mentioning this in a comment